### ISSUE TITLE
Android client crashed

### ISSUE THREAD

**Issue by SpamReceiver** [nextcloud/android#5843]
### Steps to reproduce
1. 
2. 
3. 

### Expected behaviour
- Tell us what should happen

### Actual behaviour
- Tell us what happens

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
************ CAUSE OF ERROR ************

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:354)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.IllegalArgumentException: Trying to execute a remote operation with a NULL Context
	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:125)
	at com.owncloud.android.ui.fragment.contactsbackup.ContactsBackupFragment$2.doInBackground(ContactsBackupFragment.java:227)
	at com.owncloud.android.ui.fragment.contactsbackup.ContactsBackupFragment$2.doInBackground(ContactsBackupFragment.java:215)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30110090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: 
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXS8DTC6
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Do you know when this happened?

**Comment by SpamReceiver** [comment]
unforunately, it happens every few minutes...

**Comment by tobiasKaminsky** [comment]
According from the stack trace you opened contact backup setting?
But this should be checked only once (and thus crashing only once…)

**Comment by tobiasKaminsky** [comment]
Can you force-close the app and try it again?

**Comment by SpamReceiver** [comment]
I force-closed (via Settings/Apps) the app 3 minutes ago and re-opened it. It takes unseriously long to open and to react (Android already asked me if I wanted to close the app...). However, the app didn't crash and after >5 minutes of patience it became reactive again.

I have check the contact backup setting: it's set to off.

**Comment by SpamReceiver** [comment]
So, it continues to crash.
While I am typing this, 3 more crash interrupt me...

************ CAUSE OF ERROR ************

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:354)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.SecurityException: Caller no longer running, last stopped +33s964ms because: cancelled due to doze
	at android.os.Parcel.createException(Parcel.java:1966)
	at android.os.Parcel.readException(Parcel.java:1934)
	at android.os.Parcel.readException(Parcel.java:1884)
	at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:195)
	at android.app.job.JobParameters.dequeueWork(JobParameters.java:243)
	at androidx.core.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
	at androidx.core.app.JobIntentService.dequeueWork(JobIntentService.java:640)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:481)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:359)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:83)
	at android.os.Binder.execTransact(Binder.java:739)


************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30110090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: PPR1.180610.011
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXS8DTC6

**Comment by ezaquarii** [comment]
The original stacktrace suggest that the async task callback fires when the fragment is not attached to the activity.

**Comment by tobiasKaminsky** [comment]
> Caused by: java.lang.SecurityException: Caller no longer running, last stopped +33s964ms because: cancelled due to doze

And the second is one (which?) job we run,but this fails due to doze…

**Comment by SpamReceiver** [comment]
... and this happens very often, every few minutes...
The only difference between the occurences that I can identify is the duration before it "fails to doze".

**Comment by SpamReceiver** [comment]
************ CAUSE OF ERROR ************

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:354)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.SecurityException: Caller no longer running, last stopped +189ms because: timed out while starting
	at android.os.Parcel.createException(Parcel.java:1966)
	at android.os.Parcel.readException(Parcel.java:1934)
	at android.os.Parcel.readException(Parcel.java:1884)
	at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:195)
	at android.app.job.JobParameters.dequeueWork(JobParameters.java:243)
	at androidx.core.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
	at androidx.core.app.JobIntentService.dequeueWork(JobIntentService.java:640)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:481)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:359)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:83)
	at android.os.Binder.execTransact(Binder.java:739)


************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30110090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: PPR1.180610.011
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXS8DTC6

**Comment by SpamReceiver** [comment]
Now I reinstalled the app, but things didn't improve, just change. The app keeps crashing. Here is a dump:

************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.client/com.owncloud.android.ui.activity.UploadListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.owncloud.android.db.OCUpload.setDataFixed(com.owncloud.android.files.services.FileUploader$FileUploaderBinder)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3114)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3257)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1948)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7050)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.owncloud.android.db.OCUpload.setDataFixed(com.owncloud.android.files.services.FileUploader$FileUploaderBinder)' on a null object reference
	at com.owncloud.android.ui.adapter.UploadListAdapter$UploadGroup.fixAndSortItems(UploadListAdapter.java:842)
	at com.owncloud.android.ui.adapter.UploadListAdapter$1.refresh(UploadListAdapter.java:194)
	at com.owncloud.android.ui.adapter.UploadListAdapter.loadUploadItemsFromDb(UploadListAdapter.java:718)
	at com.owncloud.android.ui.adapter.UploadListAdapter.<init>(UploadListAdapter.java:216)
	at com.owncloud.android.ui.activity.UploadListActivity.setupContent(UploadListActivity.java:176)
	at com.owncloud.android.ui.activity.UploadListActivity.onCreate(UploadListActivity.java:156)
	at android.app.Activity.performCreate(Activity.java:7327)
	at android.app.Activity.performCreate(Activity.java:7318)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3094)
	... 11 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30110090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: PPR1.180610.011
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXS8DTC6

**Comment by SpamReceiver** [comment]
However, I see many new complaints in the issue database, so I assume there has been a bigger problem introduced by most recent app updates...

**Comment by tobiasKaminsky** [comment]
@ezaquarii can you image why the OCUpload[] can contain a null entry?

**Comment by SpamReceiver** [comment]
I have configured /storage/emulated/0, i.e. the phone's internal storage, for automatic upload.

**Comment by ezaquarii** [comment]
@tobiasKaminsky 
Corrupted entries come from here:

https://github.com/nextcloud/android/blob/5d428fe9a2f7870505d462672956a83266e0b241/src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java#L286-316

First, it initializes an array, which is `null`-filled (this is how Java does it). Then it populates each field in the loop. As you can see in lines 304-307, entries that cannot be deserialized from DB model are skipped, leaving `null` in array.

I suspect that the user have suffered some data corruption. A workaround for him would be to clear the app data and login again.

**Comment by ezaquarii** [comment]
@tobiasKaminsky please cancel that; I don't see how the `createOCUploadFromCursor()` can possibly fail here.

One last desperate assumption would be to take some nasty race codition that we discovered with @fogninid back in a time when he tried to fix `OCUploadComparator` contract.

**Comment by SpamReceiver** [comment]
Maybe it's worth noting that I have MANY conflict messages in the app.
And amongst them often repeating conflict messages for the same file.

**Comment by tobiasKaminsky** [comment]
Duplicate of #5805
